### DISCUSSION TITLE
希望添加颜色自定义配置项

### DISCUSSION THREAD

**Discussion by BlakeNeko** [ppoffice/hexo-theme-icarus#888] [Q&A] [answered]
> 确保你在提交功能建议之前仔细阅读了[Hexo文档](https://hexo.io/zh-cn/)，[Icarus用户指南](https://ppoffice.github.io/hexo-theme-icarus/tags/Icarus%E7%94%A8%E6%88%B7%E6%8C%87%E5%8D%97/)，和[GitHub issues](https://github.com/ppoffice/hexo-theme-icarus/issues)来了解你的建议是否已经被他人提出过。

**你的功能建议与某个使用问题相关么？请详述。**

在配置文档里未找到所需的配置项。
在搜索issue后发现 #596 中提供的方案在4.3.0版本中不能进行颜色修改（甚至未找到对应的代码块）。

**描述你想要的解决方案**

希望可以添加针对颜色的配置选项。如：按钮颜色，文章标签颜色，超链接普通/hover状态下的颜色等。

**描述你考虑过的替代办法**

尝试以默认网页上配色“#3273dc”为关键词进行搜索，搜索结果定位在生成的静态文件的“public\css”目录下，考虑到每次执行"hexo claen && hexo g"后即使之前做过批量替换也会被还原，因此觉得不是最佳方案。

个人认为可以通过在某个文件里提供一系列颜色相关变量的配置项来修改。

**额外上下文**

无。

**Comment by ppoffice** [comment, marked as answer]
@BlakeNeko 
如果每个地方的配色都要添加单独的配置值的话，未免会使配置过于臃肿。

不过，自己动手修改主题颜色还是很简单的。请看这里：
https://github.com/ppoffice/hexo-theme-icarus/blob/4.3.0/include/style/base.styl#L15

你可以从下面这些地方找到所有可以修改的颜色，修改成你想要的值，然后把他们拷贝到上面的文件顶部就可以自定义颜色了：
https://github.com/groenroos/bulma-stylus/blob/v0.8.0/stylus/utilities/initial-variables.styl#L1-L25
https://github.com/groenroos/bulma-stylus/blob/v0.8.0/stylus/utilities/derived-variables.styl#L1-L86

**Reply by BlakeNeko** [reply]
十分感谢指点！

经个人尝试基本上只需要在`initial-variables.styl`中先添加自己想要的颜色，再更改`base.styl`中`$primary`和`derived-variables.styl`中`$link`的配置，即可应用到大多数元素上从而更改主题色。
